File: cpp/alerts/peer_error.cpp
#include "./peer_error.h"
#include "cpp/error.h"

namespace ltrs {

uint8_t peer_error_alert_get_op(lt::peer_error_alert* alert) {
    return static_cast<uint8_t>(alert->op);
}

Error peer_error_alert_get_error(lt::peer_error_alert* alert) {
    return error_code_to_error(alert->error);
}

} // namespace ltrs

File: src/ffi/alerts/peer_error.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type peer_error_alert = crate::ffi::ffi::peer_error_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/peer_error.h");
        type Error = crate::ffi::error::ffi::Error;
        unsafe fn peer_error_alert_get_op(alert: *mut peer_error_alert) -> u8;
        unsafe fn peer_error_alert_get_error(alert: *mut peer_error_alert) -> Error;
    }
}
